ui/app/src/components/CreateProjectDialog/CreateProjectDialog.tsx

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ChangeEvent, Dispatch, DispatchWithoutAction, useState } from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button, TextField, Box } from '@mui/material';
+
+export interface CreateProjectDialogProps {
+  open: boolean;
+  onClose: DispatchWithoutAction;
+  onConfirm: Dispatch<string>;
+}
+
+/**
+ * Dialog used to create a project.
+ * TODO: See with experts if the validation can't be done in a more straightforward way as it sounds complicate for
+ *   nothing. Just validating here a required field...
+ * @param props.open Define if the dialog should be opened or not.
+ * @param props.closeDialog Provides the function to close itself.
+ * @param props.onConfirm Action to perform when user confirmed.
+ * @constructor
+ */
+const CreateProjectDialog = (props: CreateProjectDialogProps) => {
+  const { open, onClose, onConfirm } = props;
+  const [projectValue, setProjectValue] = useState<string>('');
+  const [touched, setTouched] = useState(false);
+
+  const triggerOnChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setProjectValue(e.target.value);
+    setTouched(true);
+  };
+
+  // Reinitialize form for next time the dialog is opened
+  const resetForm = () => {
+    setProjectValue('');
+    setTouched(false);
+  };
+  const triggerOnConfirm = () => {
+    onClose();
+    onConfirm(projectValue);
+
+    resetForm();
+  };
+  const triggerOnClose = () => {
+    onClose();
+    resetForm();
+  };
+  return (
+    <Dialog open={open} onClose={triggerOnClose} aria-labelledby="confirm-dialog">
```

Comment:
Is it worth communicating the constraints of the title in here somewhere? It was confusing to add a title with a space in it and the only feedback I got was a popup message with "400 bad request."

Reply:
Yes I did only validation on the emptiness of the field but I can add some validation verifying the wrong characters + if already existing as well. I wanted to make sure before adding validations that the way I'm doing is correct in a React world. It sounds too complex currently.

I can also pass the action to the dialog and close on submit ONLY if the action succeeded. That would be interesting to not have to search & click again on the ADD PROJECT button if any error occurs with the HTTP request.